### DISCUSSION TITLE
User-specific locale selection?

### DISCUSSION THREAD

**Discussion by mattpr** [requarks/wiki#3133] [Help / Questions]
# Scenario:

- Default locale (for most users) should be German.  
- However specific users want to use the wiki (user interface) in their own local language (e.g. English).

It appears locale is set on a site level.  Pages can be "tagged" into different languages (to support multiple language versions of the same page).  However it doesn't appear there is a way for the user to edit their personal locale under their user profile settings (just timezone, date format, theme, etc).

I would sort of expect that user interface language preference would be treated independently (per-user basis) of the multi-language versions of a single page feature.  This is a common need especially as we have more and more distributed international teams.  Everyone might use english as a common language when communicating with colleagues, but it is nice for people to be able to use UIs in their mother tongue.

Did I miss something or is this not possible today?

If not possible, is this a change you would be open to if I get around to adding it?

Thanks!

![image](https://user-images.githubusercontent.com/1196252/108871218-361b5700-75f9-11eb-9a4e-15d6855cc885.png)

![image](https://user-images.githubusercontent.com/1196252/108871427-71b62100-75f9-11eb-9be0-465b94937f05.png)

**Comment by mattpr** [comment]
In addition to having user-interface language be customizable on a user-profile basis (separate from page reading language)...

In the multi-locale (multi-lingual pages) scenario, it would be nice if one of the languages could be designated as the default and skip the language slug in the URL.  If the common case is heavily biased towards one language (ie only a few pages have multi-lang versions), nice to keep the URLs more readable by skipping the lang slug on the default language.

This second point is somewhat answered here: https://github.com/Requarks/wiki/discussions/2994 
Basically locale slug can be disabled if you disable locale switching **AND** rebuild the site (probably what I missed when testing).  

Still it would be great to be able to exclude the language slug for the default/primary language if the site is heavily biased.  Google will still know the page language via the `<html lang="en">` attribute I would think.
